examples/template-angular-ts/src/app/app.component.ts

```diff
@@ -0,0 +1,14 @@
+import { Component } from '@angular/core';
+import { from } from 'rxjs';
+import { toggleMachine } from 'src/toggle.machine';
+import { interpret } from 'xstate';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+  actor = interpret(toggleMachine).start();
```

Comment:
shouldn't this happen in `ngOnInit`?

Reply:
Not necessarily. I would move this logic to `ngOnInit` if I had a component input that I wanted to use for the machine context for example. 

Angular TS defaults have been getting more strict. We could put the initialization logic in `ngOnInit` but then we would have to either use the non-null assertion operator, use optional chaining while accessing the actor property or initializing in the constructor which would put us where we started.

e.g.
```ts
export class AppComponent implements OnInit {
  actor!: InterpreterFrom<typeof toggleMachine>;

  ngOnInit(): void {
    this.actor = interpret(toggleMachine).start();
  }
}
```

Reply:
https://angular.io/guide/lifecycle-hooks#initializing-a-component-or-directive

> Components should be cheap and safe to construct. You should not, for example, fetch data in a component constructor. You shouldn't worry that a new component will try to contact a remote server when created under test or before you decide to display it.
> An ngOnInit() is a good place for a component to fetch its initial data.

It's not unlikely that some async jobs could start at the init time of a machine. In such a case, isn't the preferred approach to use `ngOnInit`? 

> Not necessarily. I would move this logic to ngOnInit if I had a component input that I wanted to use for the machine context for example.

It seems that `ngOnInit` is something that fits more use cases, shouldn't it be preferred here even if only for "easy to extend" argument? What I mean is that it's somewhat annoying to move the code around only because you have a new requirement - it's somewhat better to start in a place that is more flexible in the first place.

Note that I'm only thinking out loud/ask questions here - I don't necessarily mean that you have to change this if you believe that this is the right approach.

Reply:
It's just personal preference, I don't really have a strong opinion about this. I can get behind your "easy to extend" argument. Will change accordingly.

For completeness sake, from I'm modifying an example from https://angular.io/api/common/AsyncPipe:
This would be a valid use case for defining an observable property and then handling it directly with the async pipe.
This wouldn't go against the advice of making an async request on the constructor because the observable is lazy and will only be executed after the async pipe subscribes to it once the component's view has rendered.

```ts
@Component({
  selector: 'async-observable-pipe',
  template: '<div><code>observable | async</code>: Data: {{ data$ | async }}</div>'
})
export class AsyncObservablePipeComponent {
  data$ = this.http.get<{ count: number; }>('...');

  constructor(private http: HttpClient) {}
}
```

Reply:
> This wouldn't go against the advice of making an async request on the constructor because the observable is lazy and will only be executed after the async pipe subscribes to it once the component's view has rendered.

Yeah, this makes sense - it's just that in our case the `.start()` is not that lazy :p